mujoco_py/mjsim.pyx

```diff
@@ -27,6 +30,15 @@ cdef class MjSim(object):
         next ``udd_state`` after applying the user-defined dynamics. This is
         useful e.g. for reward functions that operate over functions of historical
         state.
+    substep_udd_fn : str or int or None
```

Comment:
Given that this is a substantially different implementation from udd, I would prefer to give it a different name. For example, what about substep_userdata_fn?